### PR TITLE
Check for password when pfx is wanted

### LINKED ISSUE #9: Arg checking does not force password when creating pfx cert
When a user specifies a type of pfx put omits a password the program prints a usage message but does not explain what the problems is

### PR DESCRIPTION
**Pull request by alecthegeek**
fixes #9

Check that password is defined (instead of null).

Also some minor "tidyup"

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -59,12 +59,12 @@ const options = yargs
...
-    if (argsv.type === "pfx") {
-      if (argsv.combined) {
-        throw new Error("Cannot use combine flag on PFX type certificate");
-      } else if (argsv.password == null || argsv.password.length == 0) {
-        throw new Error("Please enter a valid password for this certificate");
+  .check(function(argv, options) {
+    if (argv.type.toLowerCase() === "pfx") {
+      if (argv.combined) {
+        throw new Error("Cannot use combine flag on PFX certificate");
+      } else if ( argv.password === undefined || argv.password.length == 0) {
```
Comment (RyanReichenberg) [inline main.js]: I'm pretty sure this whole line can become:
```js
else if ( !argv.password)
```

Here is a fiddle to prove it: https://jsfiddle.net/htxad0j2/

File: main.js [hunk 2]
```diff
@@ -59,12 +59,12 @@ const options = yargs
     describe: "Sets the hostname of the subject alternate name",
     type: "string"
   })
-  .check(function(argsv) {
-    if (argsv.type === "pfx") {
-      if (argsv.combined) {
-        throw new Error("Cannot use combine flag on PFX type certificate");
-      } else if (argsv.password == null || argsv.password.length == 0) {
-        throw new Error("Please enter a valid password for this certificate");
+  .check(function(argv, options) {
```
Comment (RyanReichenberg) [inline main.js:62]: Why was the options parameter added?

Comment (alecthegeek) [reply main.js:62]: cause that is what the docs say is expected.

Comment (RyanReichenberg) [reply main.js:62]: I don't think it's needed but I guess it doesn't matter.

**Pull request merged**